src/AppContainer.tsx

```diff
@@ -93,49 +82,50 @@ function useWindowDimensions() {
 interface StateProps {
     initialLoadComplete: boolean
     extensionsConfig: ExtensionsConfig
-    workspaceInfo: WorkspaceInfo
     websocketStatus: string
 }
 
 interface DispatchProps {
-    onSetWorkspaceInfo: (workspaceInfo: WorkspaceInfo) => void
 }
 
 interface OwnProps {
+    onSetWorkspaceInfo: (workspaceInfo: WorkspaceInfo) => void
 }
 
 type Props = StateProps & DispatchProps & OwnProps & RouteComponentProps
 
-const AppContainer: FunctionComponent<Props> = ({ initialLoadComplete, workspaceInfo, onSetWorkspaceInfo, extensionsConfig, websocketStatus }) => {
-    const { workspaceName } = workspaceInfo;
-
+const AppContainer: FunctionComponent<Props> = ({ onSetWorkspaceInfo, initialLoadComplete, extensionsConfig, websocketStatus }) => {
     const { width, height } = useWindowDimensions()
 
-    if (!workspaceName) {
-        return (
-            <Switch>
-                <Route
-                    path="/:workspaceName/:path*"
-                    render={({ match, location }) => {
-                        const query = QueryString.parse(location.search);
-                        return <SetWorkspaceInfo
-                            workspaceName={match.params.workspaceName}
-                            feedUri={(query.feed as string) || null}
-                            onSetWorkspaceInfo={onSetWorkspaceInfo}
-                        />
-                    }}
-                />
-                <Route path="/"><Redirect to="/default" /></Route>
-            </Switch>
-        )
-        
+    const location = useLocation()
```

Comment:
This is the new logic for obtaining the workspaceInfo from the URL. The page URL (location) has the form:
`http://......../<workspace-name>/<page-name>?feed=<feedUri>`

Reply:
Right now we get an "Invalid page" exception. Maybe we should make it more friendly in the future.